ansible/files/whisks_design_document_for_activations_db_filters_v2.1.0.json

```diff
@@ -3,7 +3,7 @@
     "activations": {
-      "map": "function (doc) {\n  var PATHSEP = \"/\";\n  var isActivation = function (doc) { return (doc.activationId !== undefined) };\n  var summarize = function (doc) {\n    var endtime = doc.end !== 0 ? doc.end : undefined;\n    return {\n        namespace: doc.namespace,\n        name: doc.name,\n        version: doc.version,\n        publish: doc.publish,\n        annotations: doc.annotations,\n        activationId: doc.activationId,\n        start: doc.start,\n        end: endtime,\n        duration: endtime !== undefined ? endtime - doc.start : undefined,\n        cause: doc.cause,\n        statusCode: (endtime !== undefined && doc.response !== undefined && doc.response.statusCode !== undefined) ? doc.response.statusCode : undefined\n      }\n  };\n\n  var pathFilter = function(doc) {\n    for (i = 0; i < doc.annotations.length; i++) {\n      var a = doc.annotations[i];\n      if (a.key == \"path\") try {\n          var p = a.value.split(PATHSEP);\n          if (p.length == 3) {\n            return p[1] + PATHSEP + doc.name;\n          } else return doc.name;\n      } catch (e) {\n        return doc.name;\n      }\n    }\n    return doc.name;\n  }\n\n  if (isActivation(doc)) try {\n    var value = summarize(doc)\n    emit([doc.namespace+PATHSEP+pathFilter(doc), doc.start], value);\n  } catch (e) {}\n}\n",
+      "map": "function (doc) {\n  var PATHSEP = \"/\";\n  var isActivation = function (doc) { return (doc.activationId !== undefined) };\n  var summarize = function (doc) {\n    var endtime = doc.end !== 0 ? doc.end : undefined;\n    return {\n        namespace: doc.namespace,\n        name: doc.name,\n        version: doc.version,\n        publish: doc.publish,\n        annotations: doc.annotations,\n        activationId: doc.activationId,\n        start: doc.start,\n        end: endtime,\n        duration: endtime !== undefined ? endtime - doc.start : undefined,\n        cause: doc.cause,\n        statusCode: (endtime !== undefined && doc.response !== undefined && doc.response.statusCode !== undefined) ? doc.response.statusCode : undefined\n      }\n  };\n  \n  var pathFilter = function(doc) {\n    var path = undefined;\n    var binding = undefined;\n    for (i = 0; i < doc.annotations.length; i++) {\n      var a = doc.annotations[i];\n      if (a.key == \"path\") {\n        path = a.value;\n      }\n      if (a.key == \"binding\") {\n        binding = a.value;\n      }\n    }\n    try {\n      if (binding) {\n        var b = binding.split(PATHSEP)\n        return b[1] + PATHSEP + doc.name;\n      }\n      if (path) {\n        var p = path.split(PATHSEP);\n        if (p.length == 3) {\n          return p[1] + PATHSEP + doc.name;\n        } else return doc.name;  \n      }\n    } catch (e) {\n      return doc.name;\n    }\n    return doc.name;\n  }\n\n  if (isActivation(doc)) try {\n    var value = summarize(doc)\n    emit([doc.namespace+PATHSEP+pathFilter(doc), doc.start], value);\n  } catch (e) {}\n}\n",
```

Comment:
please use a new view with a different version number since updating this design doc will cause the view to recompute which can take a long time.